packages/amplify-prompts/src/__tests__/prompter.test.ts

```diff
@@ -0,0 +1,104 @@
+import { prompter } from '../prompter';
+import { prompt } from 'enquirer';
+import * as flags from '../flags';
+
+jest.mock('../flags');
+
+type Writeable<T> = { -readonly [P in keyof T]: T[P] };
+
+const flags_mock = flags as jest.Mocked<Writeable<typeof flags>>;
+
+jest.mock('enquirer', () => ({
+  prompt: jest.fn(),
+}));
+
+const prompt_mock = prompt as jest.MockedFunction<typeof prompt>;
+
+beforeEach(() => {
+  jest.clearAllMocks();
+  flags_mock.isYes = false;
+});
+
+describe('confirmContinue', () => {
+  it('returns true if yes flag set', async () => {
+    flags_mock.isYes = true;
+    expect(await prompter.confirmContinue()).toBe(true);
+    expect(prompt_mock.mock.calls.length).toBe(0);
+  });
+
+  it('returns prompt response by default', async () => {
+    prompt_mock.mockResolvedValueOnce({ result: true });
+    expect(await prompter.confirmContinue()).toBe(true);
+
+    prompt_mock.mockResolvedValueOnce({ result: false });
+    expect(await prompter.confirmContinue()).toBe(false);
+  });
+});
+
+describe('yesOrNo', () => {
+  it('returns default value if yes flag set', async () => {
+    flags_mock.isYes = true;
+    expect(await prompter.yesOrNo('test message', false)).toBe(false);
+    expect(await prompter.yesOrNo('test message', true)).toBe(true);
+  });
+
+  it('returns prompt response by default', async () => {
+    prompt_mock.mockResolvedValueOnce({ result: true });
+    expect(await prompter.yesOrNo('test message', false)).toBe(true);
+  });
+});
+
+describe('input', () => {
+  it('throws if yes flag set', async () => {
+    flags_mock.isYes = true;
+    expect(() => prompter.input('test message')).rejects.toThrowErrorMatchingInlineSnapshot(
+      `"Cannot prompt for [test message] when '--yes' flag is set"`,
+    );
+  });
+
+  it('returns prompt response if no transformer present', async () => {
+    const result = 'this is the result';
+    prompt_mock.mockResolvedValueOnce({ result });
+    expect(await prompter.input('test message')).toEqual(result);
+  });
+
+  it('returns transformed response if transformer present', async () => {
+    const promptResponse = 'this is the result';
+    const transformedValue = 'transformmed valie';
```

Comment:
typos? transformmed => transformed, valie => value

```suggestion
    const transformedValue = 'transformed value';
```